### ISSUE TITLE
Zooming into geoviews plot not working with many gv.points

### ISSUE THREAD

**Issue by Michael-Barletta** [holoviz/geoviews#647]
#### ALL software version info
bokeh                      3.1.1             pyhd8ed1ab_0    conda-forge
geopandas              0.13.0           pyhd8ed1ab_0    conda-forge
geoviews                 1.10.0           pypi_0                 pypi
holoviews                1.16.0           pyhd8ed1ab_0    conda-forge

#### Description of expected behavior and the observed behavior

Have not been able to come up with an MRE to reproduce what I am seeing, but since the latest geoviews/panel updates some of my geoviews plots create this error when zooming into them.

Before Zoom:
![image](https://github.com/holoviz/geoviews/assets/102544976/0aa5151b-ecda-4905-9785-4a404cb5b5b2)

After Zoom:
![image](https://github.com/holoviz/geoviews/assets/102544976/813ef9e8-c2e1-4dc1-8657-a36866bb3279)

Zooming in more:
![image](https://github.com/holoviz/geoviews/assets/102544976/0629ea19-06ae-43ca-b994-40b1ad535982)

This issue only occurs when zooming in with the wheel scroller, when using the box zoom, we get the expected behavior

Zooming with box zoom:
![image](https://github.com/holoviz/geoviews/assets/102544976/1cfe3d4e-d530-4438-84d5-3ac52fc4ecb9)

This seems to only occur with one plot that I have that has many different points, hence why I am struggling to reproduce the issue. Here is the behavior when zooming in with other plots that have less point data. 

Before Zoom:
![image](https://github.com/holoviz/geoviews/assets/102544976/f32891ca-bcdb-4377-a026-abcfec78056c)

After Zoom:
![image](https://github.com/holoviz/geoviews/assets/102544976/84349c55-c6a5-4377-bc85-7a09e4211a40)

**Comment by mattpap** [comment]
@Michael-Barletta, can you provide what web browser you use (including version number) and output from JS console (full traceback) when the error occurs? Given the behavior showed in the third image, there should be a lot of errors in the console.

**Comment by TheoMathurin** [comment]
I've encountered this exact problem in the past. I can't remember what fixed it, but do you happen to add a custom colorbar?

**Comment by Michael-Barletta** [comment]
@mattpap @TheoMathurin 

I am using google chrome 113.0.5672.127
here is a screenshot of the errors I am getting from the JS console as I zoom in. 
![image](https://github.com/holoviz/geoviews/assets/102544976/3f19af2a-43bc-4581-9e5b-0a6d1b8ee1ad)

interestingly enough, I am using a custom colorbar, however, I have other outputs that are using a custom colorbar but they are not producing this same issue.

**Comment by Michael-Barletta** [comment]
@Hoxbro @TheoMathurin @mattpap 
I've created the MRE and I've narrowed down the problem to the labels that are causing this issue. The only problem is that this problem seems to only pop up when my pandas dataframe is over 5000 points or so, so I can't include an example dataframe. I could however give you the exact dataframe I'm working with in .csv format, but I am not sure how to go about that.

Here is the code, when you take away the *labels argument at the very end, everything works fine, or if you have less points with the *labels argument, things are fine as well.

`import geoviews as gv
import panel as pn
import pandas as pd

gv.extension('bokeh', 'matplotlib')

df = pd.read_csv('../../data/ASOS_MPING_data/All_data.csv')
df['Time'] = pd.to_datetime(df['Time'])

kdims=['Lon','Lat','Time']

vdims = ['Station','OBS Time','Temperature (F)']

df_sel = df.loc[df["Label"] == '2m Temperature']

df_sel['Temperature (F)'] = df_sel['Temperature (F)'].astype(float)

ds = gv.Dataset(df_sel, kdims=kdims)
plot = (ds.to(gv.Points, ["Lon", "Lat"], vdims=vdims))

labels = ds.to(gv.Labels, kdims=['Lon', 'Lat'],vdims='Temperature (F)_r').opts(
text_font_size='10pt', text_color='white')

pn.panel((plot.opts(frame_width=1600, frame_height=1000, size=35))*labels)`

**Comment by Michael-Barletta** [comment]
@Hoxbro 

I updated my example with this included, I apologize, I did not realize how easy it was to generate a fake dataset that large, but that example is giving me the same issue.

**Comment by Hoxbro** [comment]
Thank you. @mattpap here is a pure Bokeh example, showing the same behavior:

``` python
import numpy as np
from bokeh.io import curdoc, show, output_notebook
from bokeh.models import ColumnDataSource, Text
from bokeh.plotting import figure

output_notebook()

N = 6000
x = np.linspace(-2, 2, N)
y = x**2
text = list(map(str, range(N)))

source = ColumnDataSource(dict(x=x, y=y, text=text))

plot = figure(title=None, width=300, height=300, min_border=0, toolbar_location=None)

glyph = Text(x="x", y="y", text="text", angle=0.3, text_color="#96deb3")
plot.add_glyph(source, glyph)

show(plot)
```
```

**Comment by mattpap** [comment]
This was fixed in https://github.com/bokeh/bokeh/pull/13113, which is available in bokeh 3.2.0.dev3.
